posthog/hogql/grammar/HogQLParser.g4

```diff
@@ -9,7 +9,9 @@ options {
 select: (selectUnionStmt | selectStmt | hogqlxTagElement) EOF;
 
 selectUnionStmt: selectStmtWithParens (UNION ALL selectStmtWithParens)*;
-selectStmtWithParens: selectStmt | LPAREN selectUnionStmt RPAREN;
+selectStmtWithParens: selectStmtWithPlaceholder | LPAREN selectUnionStmt RPAREN;
+selectStmtWithPlaceholder: selectStmt | placeholder;
```

Comment:
Is there a benefit to splitting `selectStmtWithPlaceholder` out as another node in the hierarchy? Because we could also do `selectStmtWithParens: selectStmt | LPAREN selectUnionStmt RPAREN | placeholder;`

Reply:
Great question, no real reason. Happy to change if we think this is important?

Reply:
I don't think it's critical (this works after all), but i'd definitely be nice. Fewer new nodes means fewer potential issues 🤷.